jina/flow/base.py

```diff
@@ -1071,26 +1071,28 @@ def start(self):
             if not getattr(v.args, 'external', False):
                 self.enter_context(v)
 
-        self._wait_until_all_ready()
+        import asyncio
+
+        if not asyncio.run(self._wait_until_all_ready()):
```

Comment:
This would fail in jupyter like enviornments, as the event-loop is already running and this code executes in the main process. 

We have a `jina.helper.run_async` method, which can be used to fix it. But executes `asyncio.run(func())` in a separate thread, which is (mostly) not different from keeping this as-is & running them directly on different threads.